### PR TITLE
[SYCL] Support seq_cst memory order in atomic_ref for AMDGPU targets

### PR DESCRIPTION
**Pull request by GeorgeWeb**
The AMDGPU backend supports sequential consistency ordering semantics for all atomics correctly and implements for all relevant architectures since GCN.
Updated the `acq_rel` and `seq_cst` memory order tests to build and run for CUDA devices with specifically sm_70 capabilities or on other device targets (i.e. amd-hip) that support these ordering semantics.

### CODE REVIEW

File: sycl/plugins/unified_runtime/CMakeLists.txt [hunk 1]
```diff
@@ -4,9 +4,9 @@
 # Options to override the default behaviour of the FetchContent to include UR
 # source code.
 set(SYCL_PI_UR_OVERRIDE_FETCH_CONTENT_REPO
-  "" CACHE STRING "Override the Unified Runtime FetchContent repository")
+  "https://github.com/GeorgeWeb/unified-runtime.git" CACHE STRING "Override the Unified Runtime FetchContent repository")
 set(SYCL_PI_UR_OVERRIDE_FETCH_CONTENT_TAG
-  "" CACHE STRING "Override the Unified Runtime FetchContent tag")
+  "df1701b9addb75b58e241a86ad4ec828573707a4" CACHE STRING "Override the Unified Runtime FetchContent tag")
```
Comment (sergey-semenov) [inline sycl/plugins/unified_runtime/CMakeLists.txt:9]: These should be reverted.

Comment (GeorgeWeb) [reply sycl/plugins/unified_runtime/CMakeLists.txt:9]: Yeah, gotta update that in general it's been stale for some time.

File: sycl/plugins/unified_runtime/CMakeLists.txt [hunk 2]
```diff
@@ -130,8 +130,13 @@ if(SYCL_PI_UR_USE_FETCH_CONTENT)
   )
 
   fetch_adapter_source(hip
-    ${UNIFIED_RUNTIME_REPO}
```
Comment (kbenzie) [inline sycl/plugins/unified_runtime/CMakeLists.txt:133]: This shouldn't be deleted

**Comment by GeorgeWeb** [comment]
@intel/llvm-reviewers-runtime can I get another look at this one, please. Thank you!

**Comment by GeorgeWeb** [comment]
@intel/llvm-reviewers-runtime / @sergey-semenov . Can you take another look at this one now, if possible. It is required now since https://github.com/oneapi-src/unified-runtime/pull/1415 was merged for the related tests to pass the CI. Thanks!

File: sycl/test-e2e/AtomicRef/atomic_memory_order_acq_rel.cpp [hunk 3]
```diff
@@ -1,4 +1,4 @@
-// RUN: %{build} -O3 -o %t.out -Xsycl-target-backend=nvptx64-nvidia-cuda --cuda-gpu-arch=sm_70
+// RUN: %clangxx -fsycl -fsycl-targets=%{sycl_triple} %if any-device-is-cuda %{ -Xsycl-target-backend --cuda-gpu-arch=sm_70 %} %s -o %t.out
```
Comment (aelovikov-intel) [inline sycl/test-e2e/AtomicRef/atomic_memory_order_acq_rel.cpp:1]: Why can't use use `%{build}`?

**Pull request closed without merging**